beebop/utils.py

```diff
@@ -51,3 +52,49 @@ def generate_mapping(p_hash, fs):
     # save as pickle
     with open(fs.network_mapping(p_hash), 'wb') as mapping:
         pickle.dump(cluster_component_dict, mapping)
+    return cluster_component_dict
+
+
+def delete_component_files(cluster_component_dict, fs, assign_result, p_hash):
+    """
+    poppunk generates >1100 component graph files. We only need to store those
+    files from the clusters our queries belong to.
+    """
+    queries_clusters = []
+    queries_components = []
+    for item in assign_result.values():
+        queries_clusters.append(item['cluster'])
+        queries_components.append(cluster_component_dict[str(item['cluster'])])
+    components = set(queries_components)
+    # delete redundant component files
+    keep_filenames = list(map(lambda x: f"network_component_{x}.graphml",
+                              components))
+    keep_filenames.append('network_cytoscape.csv')
+    keep_filenames.append('network_cytoscape.graphml')
+    keep_filenames.append('cluster_component_dict.pickle')
+    prev_wd = os.getcwd()
+    os.chdir(fs.output_network(p_hash))  # Change directory to network folder
+    # Loop through files in folder and remove files not in list
+    for item in os.listdir(os.getcwd()):
+        if item not in keep_filenames:
+            os.remove(item)
+    os.chdir(prev_wd)  # Change directory back to previous wd
```

Comment:
not a huge fan of this changing directories - you can avoid it if you pass the path to os.listdir and then build a path in the remove but it's not a huge problem.

If you keep it though you should put it within try/finally so that the directory is always changed back so that if there's an unexpected error nothing gets confused by the change of directory